### ISSUE TITLE
argFractional doesn't work as expected

### ISSUE THREAD

**Issue by int-index** [Gabriel439/Haskell-Turtle-Library#84]
In #83 `argFractional` was introduced to generalize `argDouble`. However, it uses `Read` instance for `Rational`, so things like `10.5` do not parse, but things like `21%2` do.

I think we should use [`scientific`](https://hackage.haskell.org/package/scientific)'s `Read` instance. `Scientific` is lossless (unlike `Double`), supports arbitrary precision (like `Rational`), and has expected syntax (unlike `Rational`).

**Comment by Gabriel439** [comment]
One solution is to `read` a `Double` and then generalize the type using `realToFrac`

`scientific` would also work, but it seems like a large dependency for a very small gain

**Comment by int-index** [comment]
```haskell
λ> read "1.33333333333333335" :: Double
1.3333333333333333
λ> read "1.33333333333333335" :: Scientific
1.33333333333333335
```
Avoiding data loss is a big gain to me. Imagine a user who tries to parse a `Scientific` with `argFractional` and this happens. In my opinion it's a bug.

If you don't want to depend on `scientific`, it's better to use `argDouble` and remove `argFractional`, because then data loss is explicit.

**Comment by Gabriel439** [comment]
Oh, I see the problem now.  In that case I'll just specialize the type to `Double`, then.

**Comment by Gabriel439** [comment]
Actually, it just occurred to me that since we have `argRead`, we don't really need `argIntegral` or `argFractional`.  The user can just use `argRead` and specialize it to the type they want.  So if the user wants to read a `Double` they just use `argRead` specialized to `Double`s.  Similarly, if they want greater precision they just specialize `argRead` to `Scientific`.  That allows us to get the best of both worlds without incurring a dependency on the `scientific` library.

In other words, the API would just have `{arg|opt}{|Read}` and `{arg|opt}{|Text}`.  What do you think?

**Comment by int-index** [comment]
I originally included monomorphic versions for two reasons:

1. Abstraction. The user doesn't need to know that the `Read` instance is used under the hood. To parse an integer one can use `argInteger`, the `Read` instance is more of an implementation detail, especially since the `Read`/`Show` hierarchy isn't about parsing and pretty printing, it's about marshalling between Haskell values and text (thus good `Show` instances produce valid Haskell code).

2. Convenience. Using a monomorphic functions is nicer than specifying types explicitly or relying on defaulting to pick an instance for parsing. In `f <$> argRead` where `f :: Ord a => a -> Bool`, it is ambiguos what the `a` type variable should be instantiated to. To specify the type explicitly the user would be forced to write something like `f <$> (argRead :: Parser Double)` or `f <$> (argRead :: Parser Integer)`. I would be ok with this if not for the redundant `Parser` type. It's possible, however, to use `PartialTypeSignatures` to solve this issue and write `f <$> (argRead :: _ Double)`.

So if we're going to remove `argIntegral` and `argFractional` instead of fixing them, let's at least document the caveats of using `argRead` so the user knows when to specify a type signature and how to do so.

**Comment by Gabriel439** [comment]
In that case I'll keep the monomorphic versions for nicer type inference and just specialize them to `Int`/`Integer`/`Double` for now and let the user define their own monomorphic versions for any other types they might want using `argRead`

**Comment by Gabriel439** [comment]
Fixed by #87
